recipes/openfst/all/conanfile.py

```diff
@@ -0,0 +1,212 @@
+        if self.settings.os != "Linux":
+            raise ConanInvalidConfiguration(
+                "OpenFst is only supported on linux")
+
+        compilers = {
+            "gcc": "8",
+            "clang": "7",
+        }
+
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, 17)
+        minimum_compiler = compilers.get(str(self.settings.compiler))
+        if minimum_compiler:
+            if tools.Version(self.settings.compiler.version) < minimum_compiler:
+                raise ConanInvalidConfiguration(f"{self.name} requires c++17, which your compiler does not support.")
+        else:
+            self.output.warn(f"{self.name} requires c++17, but this compiler is unknown to this recipe. Assuming your compiler supports c++17.")
+
+        # Check stdlib ABI compatibility
+        if self.settings.compiler == "gcc" and self.settings.compiler.libcxx != "libstdc++11":
+            raise ConanInvalidConfiguration('Using %s with GCC requires "compiler.libcxx=libstdc++11"' % self.name)
+        elif self.settings.compiler == "clang" and self.settings.compiler.libcxx not in ["libstdc++11", "libc++"]:
+            raise ConanInvalidConfiguration('Using %s with Clang requires either "compiler.libcxx=libstdc++11"'
+                                            ' or "compiler.libcxx=libc++"' % self.name)
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+    @functools.lru_cache(1)
+    def _configure_autotools(self):
+        autotools = AutoToolsBuildEnvironment(self)
+        yes_no = lambda v: "yes" if v else "no"
+        args = [
+            "--with-pic={}".format(yes_no(self.options.get_safe("fPIC", True))),
+            "--enable-shared={}".format(yes_no(self.options.shared)),
+            "--enable-static={}".format(yes_no(not self.options.shared)),
+            "--enable-bin={}".format(yes_no(self.options.enable_bin)),
+            "--enable-compact-fsts={}".format(yes_no(self.options.enable_compact_fsts)),
+            "--enable-compress={}".format(yes_no(self.options.enable_compress)),
+            "--enable-const-fsts={}".format(yes_no(self.options.enable_const_fsts)),
+            "--enable-far={}".format(yes_no(self.options.enable_far)),
+            "--enable-grm={}".format(yes_no(self.options.enable_grm)),
+            "--enable-linear-fsts={}".format(yes_no(self.options.enable_linear_fsts)),
+            "--enable-lookahead-fsts={}".format(yes_no(self.options.enable_lookahead_fsts)),
+            "--enable-mpdt={}".format(yes_no(self.options.enable_mpdt)),
+            "--enable-ngram-fsts={}".format(yes_no(self.options.enable_ngram_fsts)),
+            "--enable-pdt={}".format(yes_no(self.options.enable_pdt)),
+            "--enable-special={}".format(yes_no(self.options.enable_special)),
+            "LIBS=-lpthread",
+        ]
+        if tools.get_env("CC"):
```

Comment:
I think these lines for CC/CXX handlings might be unnecessary, does it fail for you without them?

Reply:
will remove that. It works without them